a3p-integration/proposals/z:acceptance/test-lib/sync-tools.js

```diff
@@ -0,0 +1,285 @@
+      log('RESULT', result);
+      if (condition(result)) {
+        return result;
+      }
+    } catch (error) {
+      if (error instanceof Error) {
+        log(`Error: ${error.message}`);
+      } else {
+        log(`Unknown error: ${String(error)}`);
+      }
+    }
+
+    retries++;
+    console.log(
+      `Retry ${retries}/${maxRetries} - Waiting for ${retryIntervalMs}ms for ${message}...`,
+    );
+    await sleep(retryIntervalMs, { log, setTimeout });
+  }
+
+  throw Error(`${message} condition failed after ${maxRetries} retries.`);
+};
+
+/**
+ * Making sure a core-eval resulted succesfully deploying a contract
+ */
+const makeGetInstances = follow => async () => {
+  const instanceEntries = await follow(
+    '-lF',
+    `:published.agoricNames.instance`,
+  );
+
+  return Object.fromEntries(instanceEntries);
+};
+
+/**
+ *
+ * @param {string} contractName
+ * @param {{follow: () => object, setTimeout: (object) => void}} ambientAuthroity
+ * @param {RetyrOptions} options
+ * @returns
+ */
+export const waitUntilContractDeplyed = (
+  contractName,
+  ambientAuthroity,
+  options,
+) => {
+  const { follow, setTimeout } = ambientAuthroity;
+  const getInstances = makeGetInstances(follow);
+  const {
+    maxRetries = 6,
+    retryIntervalMs = 3500,
+    log = console.log,
+    errorMessage = 'Error',
+  } = options;
+
+  return retryUntilCondition(
+    getInstances,
+    instanceObject => Object.keys(instanceObject).includes(contractName),
+    errorMessage,
+    // @ts-ignore
+    { maxRetries, retryIntervalMs, log, setTimeout },
+  );
+};
+
+// Making sure an account is successfully funded with vbank assets like IST, BLD etc.
+
+const makeQueryCosmosBalace = queryCb => async dest => {
+  const conins = await queryCb('bank', 'balances', dest);
+  return conins.balances;
+};
+
+/**
+ *
+ * @param {Array} balances
+ * @param {CosmosBalanceThresold} thresold
+ * @returns {boolean}
+ */
+const checkCosmosBalance = (balances, thresold) => {
+  const balance = [...balances].find(({ denom }) => denom === thresold.denom);
+  return Number(balance.amount) >= thresold.value;
+};
+
+/**
+ * @param {string} destAcct
+ * @param {{query: () => Promise<object>, setTimeout: (object) => void}} ambientAuthroity
+ * @param {{denom: string, value: number}} threshold
+ * @param {RetyrOptions} options
+ * @returns
+ */
+export const waitUntilAccountFunded = (
+  destAcct,
+  ambientAuthroity,
+  threshold,
+  options,
+) => {
+  const { query, setTimeout } = ambientAuthroity;
+  const queryCosmosBalance = makeQueryCosmosBalace(query);
+  const {
+    maxRetries = 6,
+    retryIntervalMs = 3500,
+    log = console.log,
+    errorMessage = 'Error',
+  } = options;
+
+  return retryUntilCondition(
+    async () => queryCosmosBalance(destAcct),
+    balances => checkCosmosBalance(balances, threshold),
+    errorMessage,
+    // @ts-ignore
+    { maxRetries, retryIntervalMs, log, setTimeout },
+  );
+};
+
+// Making sure an offer is resulted
```

Comment:
If this comment is intended as a marker to set off a section of the file (which is a practice I support, though it's not required) I prefer something more prominent. I usually use something like this
```suggestion
// ///////////// Making sure offers get results ///////////////
```

Reply:
resolved in https://github.com/Agoric/agoric-sdk/pull/10171/commits/d3f245300f2d42b83bc761c1e22950feeacd63db